From 8f4e5a41c58986b68406251051c26beb059dd9ce Mon Sep 17 00:00:00 2001
From: Tobias Fella <fella@posteo.de>
Date: Sun, 28 May 2023 11:22:40 +0200
Subject: [PATCH] Drop reset*Count invocations

Still coming from Spectral, where they were taken from Quaternion, where they were dropped because they don't work as expected.
Also, soon to be removed from libQuotient itself.
---
 src/neochatroom.cpp | 10 ----------
 src/neochatroom.h   |  1 -
 2 files changed, 11 deletions(-)

diff --git a/src/neochatroom.cpp b/src/neochatroom.cpp
index a98b6cdbe..101c0fc50 100644
--- a/src/neochatroom.cpp
+++ b/src/neochatroom.cpp
@@ -66,8 +66,6 @@ NeoChatRoom::NeoChatRoom(Connection *connection, QString roomId, JoinState joinS
     : Room(connection, std::move(roomId), joinState)
 {
     connect(connection, &Connection::accountDataChanged, this, &NeoChatRoom::updatePushNotificationState);
-    connect(this, &NeoChatRoom::notificationCountChanged, this, &NeoChatRoom::countChanged);
-    connect(this, &NeoChatRoom::highlightCountChanged, this, &NeoChatRoom::countChanged);
     connect(this, &Room::fileTransferCompleted, this, [this] {
         setFileUploadingProgress(0);
         setHasFileUploading(false);
@@ -410,14 +408,6 @@ void NeoChatRoom::onRedaction(const RoomEvent &prevEvent, const RoomEvent & /*af
     }
 }
 
-void NeoChatRoom::countChanged()
-{
-    if (displayed() && !hasUnreadMessages()) {
-        resetNotificationCount();
-        resetHighlightCount();
-    }
-}
-
 QDateTime NeoChatRoom::lastActiveTime()
 {
     if (timelineSize() == 0) {
diff --git a/src/neochatroom.h b/src/neochatroom.h
index d0795f383..6125cfa96 100644
--- a/src/neochatroom.h
+++ b/src/neochatroom.h
@@ -866,7 +866,6 @@ private:
 #endif
 
 private Q_SLOTS:
-    void countChanged();
     void updatePushNotificationState(QString type);
 
     void cacheLastEvent();
-- 
GitLab

